flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -424,6 +424,27 @@ public boolean isShutdown() {
 		}
 	}
 
+	public CompletableFuture<CompletedCheckpoint> triggerSynchronousCheckpoint(
```

Comment:
It seems to me that we could reuse the existing codes of `triggerSynchronousSavepoint`, adding a flag to indicate whether it's a checkpoint or savepoint. The same applies to `LegacyScheduler#stopWithCheckpoint`